### DISCUSSION TITLE
Create Multiple Models in one command

### DISCUSSION THREAD

**Discussion by ousid** [laravel/framework#40718] [Ideas] [answered]
I have been thinking lately on this idea, and on creating a `PR` for it, if it's fit.

Creating Multiple Models in one command like:
```bash
php artisan make:model Product Order Invoice
```
With this it creates the following model all at once, and maybe can add flags `--all` or `--controller` ...etc. It creates each file for each model.

The question is: Is it a good Idea to create a PR for it, or it's just a waste of time?

Hints on this, would be much appreciated.

**Comment by chuckrincon** [comment, marked as answer]
> The question is: Is it a good Idea to create a PR for it, or it's just a waste of time?

It's a good idea for a PR but probably it will be rejected for adding unnecessary complexity to the command.

You could use `bash` for achieve the same results:

Using the flag `--all`:
```bash
for name in {Product,Order,Invoice}; do php artisan make:model -a "$name"; done
```
Or using the flags `--migration`, `--factory`, `--seed` and `--controller`:
```bash
for name in {Product,Order,Invoice}; do php artisan make:model -m -f -s -c "$name"; done
```
Alternatively you could create a package to offer this functionality, with a different `make` command:
```bash
php artisan make:models --all Product Order Invoice
```

**Reply by ousid** [reply]
Hey, @chuckrincon 

I created a [package](https://github.com/coderflexx/laracommand) to create multiple models as mentioned above, give it a look. 
I'll be happy if there are any notes :)
thanks.

**Comment by ousid** [comment]
Thanks, @chuckrincon for your detailed answer.

Yeah, the package idea would be great, instead of adding complexity to the command itself, especially the newcomers.

I'll work on a package as soon as I can.
